src/pages/index.js

```diff
@@ -74,15 +108,60 @@ function handleAddCardSubmit(inputValues) {
     link: inputValues.url,
   };
 
-  const cardView = createCard(cardData);
-
-  cardSection.addItem(cardView);
+  addModal.setLoading(true);
+  api
+    .addCard(cardData)
+    .then((res) => {
+      cardSection.addItem(createCard(res));
+      addModal.close();
+      addCardFormElement.reset();
+      addCardValidator.disableButton();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      addModal.setLoading(false);
+    });
+}
 
-  addModal.close();
-  addCardFormElement.reset();
-  addCardValidator.disableButton();
+function handleLikeClick(card) {
+  if (card._isLiked) {
+    api.dislikeCard(card._id).then(() => {
+      card.toggleLikeIcon();
+    });
+  } else {
+    api.likeCard(card._id).then(() => {
+      card.toggleLikeIcon();
+    });
+  }
 }
 
+const avatarModal = new ModalWithForm("#avatar-modal", ({ url }) => {
+  const avatarForm = document.querySelector("#avatar-form");
```

Comment:
`  const avatarForm = document.querySelector("#avatar-form");
`

It’s better to find **constant DOM elements** only 1 time at the top of the file so as not to waste resources on searching them again and again when you call a method (function) because searching is a very **hard** operation for the browser engine